### PR TITLE
Update community maintenance doc

### LINKED ISSUE #3305: devstats: many new repos aren't included in devstats
ref: https://github.com/cncf/devstats/issues/158

I ran a script to generate the SIG-based grouping last time, but of course like a genius I don't think I actually checked it in.  Creating the issue here so it can be /assign'ed and maybe handled by one of the other devstats subproject owners.  I'm not sure when I'll be able to get to this.

/area devstats
/sig contributor-experience
/priority important-longterm

/assign @jberkus @Phillels @spiffxp 
subproject owners I can assign to

### PR DESCRIPTION
**Pull request by mrbobbytables**
fixes: #3305
fixes: #4477 

- Removed unnecessary checks from slack maintenance tasks
  - Follow up on changes discussed in #4477
- Updated links to PR/Issue templates
- Removed outdated gsuite audit task
  - With the automation of group management in k8s.io, a manual check is no longer necessary.
- Add devstats task regarding updating repo groups
  - While this does not add automation to mapping repo -> repo group I think it's still safe to close #3305. With the new dashboard it is significantly easier to get an idea of what needs to be updated at a glance and the majority of changes here forward should be pretty minimal.

/cc @nikhita @castrojo @jberkus

### CODE REVIEW

File: sig-contributor-experience/community-maintenance.md [hunk 1]
```diff
@@ -1,45 +1,53 @@
...
 
 - [ ] Channels - Anything out of place or not following the guidelines? 
-  - [ ] Close unused channels. 
-  - [ ] Check to see if there is a purpose, pinned documents like agendas, and other best practices being used in the channel. These are not required but are useful.
+  - [ ] For community group channels - Check to see if there is a purpose,
+        pinned documents like agendas, and other best practices being used in
+        the channel. These are not required but are useful.
 - [ ] Audit the Integrations and apps
 - [ ] Audit the user tokens
 - [ ] Audit the Emojis
```
Comment (jberkus) [inline sig-contributor-experience/community-maintenance.md]: So these three items have been automated already (thanks @Katharine).  Is there value in having them on this checklist?  I almost feel like we need a separate list of "things that are automated and we would be sad if they went away, but we don't need to actively pay attention to"

File: sig-contributor-experience/community-maintenance.md [hunk 2]
```diff
@@ -1,45 +1,53 @@
...
 - [ ] Generate a Slack data report to share with the community (optional)
```
Comment (jberkus) [inline sig-contributor-experience/community-maintenance.md]:
```suggestion
- [ ] Generate a Slack data report to share with the community (optional)
- [ ] Audit list of Moderators Pro Tempore and remove those no longer available (and hopefully replace them)
- [ ] Rotate full-time Slack moderators every 3-6 months 
```

File: sig-contributor-experience/community-maintenance.md (hunk 1 above)
Comment (mrbobbytables) [reply sig-contributor-experience/community-maintenance.md]: We can probably remove the emoji and integrations checks, the user-tokens one we should still keep an eye on and ping folks about them -- like @mariantalla was using one to run a report but that's not likely needed to remain permanently.

**Comment by mrbobbytables** [comment]
Should have tossed a hold on this for review, but can follow-up with more changes.

**Pull request merged**